### PR TITLE
docs: reflect v2.3 release changes in roadmap.md

### PR DESCRIPTION
**Pull request by alexmt**
Signed-off-by: Alexander Matyushentsev <AMatyushentsev@gmail.com>

PR reflects v2.3 release changes in roadmap.md .

### CODE REVIEW

**Comment by alexmt** [comment]
Based on discussing in contributor meetings I've moved several items to v2.4.  @wtam2018 , @leoluz , @crenshaw-dev  let me know please if you want to add/remove something from v2.4 plans.

File: docs/roadmap.md [hunk 1]
```diff
@@ -1,26 +1,28 @@
...
     - [Config Management Tools Integrations UI/CLI](#config-management-tools-integrations-uicli)
+  - [v2.5 and beyond](#v25-and-beyond)
+    - [First class support for ApplicationSet resources](#first-class-support-for-applicationset-resources)
+    - [Merge Argo CD Image Updater into Argo CD](#merge-argo-cd-image-updater-into-argo-cd)
+    - [Support multiple sources for an Application](#support-multiple-sources-for-an-application)
+    - [Sharding application controller](#sharding-application-controller)
+    - [Add support for secrets in Application parameters](#add-support-for-secrets-in-application-parameters)
     - [Allow specifying parent/child relationships in config](#allow-specifying-parentchild-relationships-in-config)
     - [Dependencies between applications](#dependencies-between-applications)
     - [Multi-tenancy improvements](#multi-tenancy-improvements)
```
Comment (jannfis) [inline docs/roadmap.md:18]: I don't want to push this one out much further, to be honest.

Comment (alexmt) [reply docs/roadmap.md:18]: I know it took forever to make progress here. Would you change your mind if we prioritize it and started working on it in e.g. next release? Meaning you won't not have to chase maintainers and wait for feedback on proposal? :)

File: docs/roadmap.md [hunk 2]
```diff
@@ -1,26 +1,28 @@
...
-    - [Maintain difference in cluster and git values for specific fields](#maintain-difference-in-cluster-and-git-values-for-specific-fields)
-    - [ARM images and CLI binary](#arm-images-and-cli-binary)
+  - [v2.4](#v24)
     - [Server side apply](#server-side-apply)
-  - [v2.4 and beyond](#v24-and-beyond)
-    - [First class support for ApplicationSet resources](#first-class-support-for-applicationset-resources)
     - [Input Forms UI Refresh](#input-forms-ui-refresh)
-    - [Merge Argo CD Image Updater into Argo CD](#merge-argo-cd-image-updater-into-argo-cd)
     - [Web Shell](#web-shell)
     - [Helm values from external repo](#helm-values-from-external-repo)
```
Comment (ishitasequeira) [inline docs/roadmap.md:8]: As we are considering to add this as part of `[Support multiple sources for an Application](#support-multiple-sources-for-an-application)`, should we move this to `v2.5` as well?

Comment (ishitasequeira) [inline docs/roadmap.md:8]: As we are considering to add this as part of [Support multiple sources for an Application](#support-multiple-sources-for-an-application), should we move this to v2.5 as well?

Comment (ishitasequeira) [inline docs/roadmap.md:8]: As we are considering to add this as part of `[Support multiple sources for an Application](#support-multiple-sources-for-an-application)`, should we move this to v2.5 as well?

Comment (alexmt) [reply docs/roadmap.md:8]: Agree. I forgot about this feature. Moved to it.

File: docs/roadmap.md [hunk 3]
```diff
@@ -1,26 +1,28 @@
...
-    - [Add support for secrets in Application parameters](#add-support-for-secrets-in-application-parameters)
     - [Config Management Tools Integrations UI/CLI](#config-management-tools-integrations-uicli)
```
[the remark this answers is not in the document]
Comment (alexmt) [reply docs/roadmap.md]: Sounds good. I've mentioned both UI/CLI and security enhancements in this roadmap item. Let me know what do you think please

[the remark this answers is not in the document]
Comment (leoluz) [reply docs/roadmap.md]: Looks good. Tks!

File: docs/roadmap.md [hunk 4]
```diff
@@ -1,26 +1,28 @@
...
+  - [v2.4](#v24)
     - [Server side apply](#server-side-apply)
-  - [v2.4 and beyond](#v24-and-beyond)
-    - [First class support for ApplicationSet resources](#first-class-support-for-applicationset-resources)
     - [Input Forms UI Refresh](#input-forms-ui-refresh)
-    - [Merge Argo CD Image Updater into Argo CD](#merge-argo-cd-image-updater-into-argo-cd)
     - [Web Shell](#web-shell)
     - [Helm values from external repo](#helm-values-from-external-repo)
+    - [Support multiple sources for an Application](#support-multiple-sources-for-an-application)
+    - [Config Management Tools Enhancements: Integrations UI/CLI & CMP Security Improvements](#config-management-tools-enhancements-integrations-uicli--cmp-security-improvements)
```
Comment (crenshaw-dev) [inline docs/roadmap.md:10]: I think parameterization and security should target 2.4 but UI/CLI integration should wait for 2.5.

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]

**Review by crenshaw-dev — APPROVED** [review]

**Pull request merged**